python/pysrc/kaskada/udf.py

```diff
@@ -1,57 +1,63 @@
 """Functionality for calling Python UDFs from Kaskada."""
+from __future__ import annotations
+
 import functools
 from typing import Callable
 
 import pandas as pd
 import pyarrow as pa
 
+from . import _ffi
+from ._timestream import Literal
+from ._timestream import Timestream
+
 
 # TODO: Allow functions to return `pd.DataFrame` for struct arrays.
 FuncType = Callable[..., pd.Series]
 
 
-class Udf(object):
-    """Class wrapping a UDF used in Kaskada."""
-
-    def __init__(self, name, func: FuncType, signature: str) -> None:
-        """Create a UDF for a function returning a Pandas series.
-
-        Parameters
-        ----------
-        name : str
-            Name of the function being wrapped.
-
-        func : FuncType
-            The callable to wrap.
+class Udf:
+    """User defined function."""
 
-        signature : str
-            The Kaskada function signature for this UDF. Will be used to check
-            types of calls to this function and propagate type information for
-            the rest of the query.
-        """
+    def __init__(self, signature: str, func: Callable[..., pa.Array]) -> None:
+        """Create a UDF with the given signature."""
         functools.update_wrapper(self, func)
```

Comment:
I suspect this is why you're not getting the docstring through -- I think you need to pass the original function, not the wrapped function -- this basically tells `functools` that "self is a wrapper around `func`", but that's the lambda we made.

You can probably move this out of the `__init__` and call `functools.update_wrapper` below.